src/WebAuthn.sol

```diff
@@ -146,19 +155,90 @@ library WebAuthn {
+    /// This ensures that on RIP-7212 chains, simulation goes through the precompile success path
+    /// (and on non-7212 chains, through the software verifier success path) without depending on
+    /// the caller's inputs, which are necessarily invalid during bundler estimation.
+    ///
+    /// All non-signature checks and hashing are still performed to approximate the same gas profile
+    /// as the normal verify function, but their outcomes are not used to determine the final return value.
+    ///
+    /// IMPORTANT: Only use this in simulation-only bytecode overrides.
+    function verifySim(
+        bytes memory challenge,
+        bool requireUV,
+        WebAuthnAuth memory webAuthnAuth,
+        uint256,
+        /* x */
+        uint256 /* y */
+    )
+        internal
+        view
+        returns (bool)
+    {
+        // Compute the same work as verify() to keep gas parity, but ignore outcomes.
+        if (webAuthnAuth.s > _P256_N_DIV_2) {
+            // no-op: do not early return; preserve computation parity
+        }
+
+        string memory _type = webAuthnAuth.clientDataJSON.slice(webAuthnAuth.typeIndex, webAuthnAuth.typeIndex + 21);
+        if (keccak256(bytes(_type)) != _EXPECTED_TYPE_HASH) {
+            // no-op
+        }
+
+        bytes memory expectedChallenge = bytes(string.concat('"challenge":"', Base64.encodeURL(challenge), '"'));
+        string memory actualChallenge =
+            webAuthnAuth.clientDataJSON.slice(webAuthnAuth.challengeIndex, webAuthnAuth.challengeIndex + expectedChallenge.length);
+        if (keccak256(bytes(actualChallenge)) != keccak256(expectedChallenge)) {
+            // no-op
+        }
+
+        if (webAuthnAuth.authenticatorData[32] & _AUTH_DATA_FLAGS_UP != _AUTH_DATA_FLAGS_UP) {
+            // no-op
+        }
+        if (requireUV && (webAuthnAuth.authenticatorData[32] & _AUTH_DATA_FLAGS_UV) != _AUTH_DATA_FLAGS_UV) {
+            // no-op
+        }
+
+        // Hashing steps to mirror the normal path's cost profile.
+        bytes32 clientDataJSONHash = sha256(bytes(webAuthnAuth.clientDataJSON));
         bytes32 messageHash = sha256(abi.encodePacked(webAuthnAuth.authenticatorData, clientDataJSONHash));
-        bytes memory args = abi.encode(messageHash, webAuthnAuth.r, webAuthnAuth.s, x, y);
-        // try the RIP-7212 precompile address
-        (bool success, bytes memory ret) = _VERIFIER.staticcall(args);
+        // Prevent solidity optimizations from pruning the above by creating a meaningless dependency.
+        if (messageHash == bytes32(0)) {
+            // no-op branch that is expected to be false for realistic inputs
+        }
+
+        // Final signature verification uses the internal fixed vector, not the caller inputs.
+        // This will succeed on both precompile-enabled and software paths, aligning simulation with onchain success.
+        return _verifySigP256(_SIM_MSG, _SIM_R, _SIM_S, _SIM_X, _SIM_Y);
```

Comment:
The most important change on this branch, hard-coding the highest-gas, known-valid experimental test vector.